src/Components/ClustersListTable/ClustersListTable.spec.ct.js

```diff
@@ -295,170 +372,158 @@ describe('clusters list table', () => {
-      cy.get(TOOLBAR_FILTER)
-        .find('.pf-c-select__menu')
-        .find('input')
-        .eq(0)
-        .click()
-        .then(() => expect(window.location.search).to.not.contain(`hits=`));
-      cy.get(TOOLBAR_FILTER)
-        .find('.pf-c-select__menu')
-        .find('input')
-        .eq(3)
-        .click()
-        .then(() => expect(window.location.search).to.contain(`hits=2`));
-      cy.get('.pf-c-table__sort').eq(4).click();
-      cy.getFirstRow().find('td[data-label=Moderate]').should('have.text', 3);
-      cy.get('.pf-c-table__sort').eq(4).click();
-      cy.getFirstRow().find('td[data-label=Moderate]').should('have.text', 19);
+    it('empty state is displayed when filters do not match any rule', () => {
+      removeAllChips();
+      filterApply({
+        name: 'Not existing clusters',
+        risk: ['Critical', 'Moderate'],
+      });
+      // TODO check empty table view
+      // TODO headers are displayed
     });
 
-    it('can filter clusters by the total risk Low', () => {
-      cy.get('.ins-c-primary-toolbar__filter button').click();
-      cy.get(TOOLBAR_FILTER)
-        .find('.pf-c-dropdown__menu')
-        .find('li')
-        .eq(1)
-        .click();
-      cy.get(TOOLBAR_FILTER).find('.pf-c-select__toggle').click();
-      cy.get(TOOLBAR_FILTER)
-        .find('.pf-c-select__menu')
-        .find('input')
-        .eq(0)
-        .click()
-        .then(() => expect(window.location.search).to.not.contain(`hits=`));
-      cy.get(TOOLBAR_FILTER)
-        .find('.pf-c-select__menu')
-        .find('input')
-        .eq(4)
-        .click()
-        .then(() => expect(window.location.search).to.contain(`hits=1`));
-      cy.get('.pf-c-table__sort').eq(5).click();
-      cy.getFirstRow().find('td[data-label=Low]').should('have.text', 1);
-      cy.get('.pf-c-table__sort').eq(5).click();
-      cy.getFirstRow().find('td[data-label=Low]').should('have.text', 14);
+    describe('single filter', () => {
+      Object.entries(filtersConf).forEach(([k, v]) => {
+        v.values.forEach((filterValues) => {
+          it(`${k}: ${filterValues}`, () => {
+            const filters = {};
+            filters[k] = filterValues;
+            let sortedNames = _.map(
+              _.cloneDeep(filterData(namedClustersDefaultSorting, filters)),
+              'name'
+            );
+            removeAllChips();
+            filterApply(filters);
+            if (sortedNames.length === 0) {
+              // TODO check empty table view
+              // TODO headers are displayed
+            } else {
+              cy.get(`td[data-label="Name"]`)
+                .then(($els) => {
+                  return _.map(Cypress.$.makeArray($els), 'innerText');
+                })
+                .should(
+                  'deep.equal',
+                  sortedNames.slice(
+                    0,
+                    Math.min(DEFAULT_ROW_COUNT, sortedNames.length)
```

Comment:
Maybe I'm wrong but I think it is needed. Let's say you have 24 rows in the UI and 24 items in `sortedNames` arrays, and only 20 displayed. When I get the innerText, I get only the first 20 and I need to compare it with a slice of 20. If I remove the min, it will compare with a slice of 24 and they won't match. I am understanding something wrong?

Reply:
yeah I should have concretized what to replace with:
```suggestion
                    DEFAULT_ROW_COUNT
```
this should work in all the cases I think.. if you've got 24 rows, it will get only 20 if this constant set to 20; otherwise, if you have 17 rows, you will get 17 because of the behavior of `slice`

Reply:
Yes, absolutely. You are totally right